tests/functional/test_rose_stem.py

```diff
@@ -106,6 +106,31 @@ def monkeymodule():
     mpatch.undo()
 
 
+@pytest.fixture(scope='class')
+def mock_usr_cfg(monkeymodule):
+    """Mock the rose ResourceLocator.default
+
+    Args (To _inner):
+        key: List of keys to mock getting.
+        value: Value to return.
+    """
+    def _inner(keys, value):
+        """Mock a config object with a config node."""
+        mock_config = type(
+            'MockConfig', (object,), {
+                'get_conf': type(
+                    'MockConfigNode', (object,),
+                    {'get_value': lambda *_: value})
+            }
+        )
+
+        monkeymodule.setattr(
+            'cylc.rose.stem.ResourceLocator.default',
```

Comment:
As with the Cylc version, this Python path may need to be edited for future use cases as `ResourceLocator` is imported from other locations within the Rose library.

Reply:
And it would make a great deal of sense to generalize this fixture, as and when we need it. But why increase the scope now?